Casks/o/oracle-jdk-javadoc.rb

```diff
@@ -11,10 +11,17 @@
   homepage "https://www.oracle.com/java/technologies/downloads/"
 
   livecheck do
-    url "https://www.oracle.com/java/technologies/javase-jdk#{version.major}-doc-downloads.html"
+    url "https://www.oracle.com/java/technologies/downloads/"
     regex(%r{/(\d+(?:\.\d+)*)(?:\+|%2B)(\d+(?:\.\d+)*)/(\h+)/jdk[._-]v?(\d+(?:\.\d+)*)_doc-all\.zip}i)
     strategy :page_match do |page, regex|
-      page.scan(regex).map { |match| "#{match[0]},#{match[1]},#{match[2]}" }
+      major = page.scan(%r{href=.*?/technologies/javase-jdk(\d+)-doc-downloads\.html}i)&.map do |match|
+        match[0].to_i
+      end&.max
+      next if major.blank?
+
+      url = "https://www.oracle.com/java/technologies/javase-jdk#{major}-doc-downloads.html"
+      version_page = Homebrew::Livecheck::Strategy.page_content(url)
+      version_page[:content]&.scan(regex)&.map { |match| "#{match[0]},#{match[1]},#{match[2]}" }
```

Comment:
```suggestion
      major = page.scan(%r{href=.*?/technologies/javase-jdk(\d+)-doc-downloads\.html}i)
                  .max_by { |match| Version.new(match[0]) }
                  &.first
      next if major.blank?

      download_page = Homebrew::Livecheck::Strategy.page_content(
        "https://www.oracle.com/java/technologies/javase-jdk#{major}-doc-downloads.html",
      )
      next if (download_page_content = download_page[:content]).blank?

      download_page_content.scan(regex).map { |match| "#{match[0]},#{match[1]},#{match[2]}" }
```

Same as https://github.com/Homebrew/homebrew-cask/pull/205516#discussion_r2002189266